cli/pkg/crd2apiresourceschema/plugin/crd2apiresourceschema.go

```diff
@@ -0,0 +1,198 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/component-base/logs"
+	logsv1 "k8s.io/component-base/logs/api/v1"
+
+	"github.com/kube-bind/kube-bind/cli/pkg/kubectl/base"
+	kubebindv1alpha2 "github.com/kube-bind/kube-bind/sdk/apis/kubebind/v1alpha2"
+)
+
+type CRD2APIResourceSchemaOptions struct {
+	Options *base.Options
+	Logs    *logs.Options
+	Print   *genericclioptions.PrintFlags
+}
+
+func NewCRD2APIResourceSchemaOptions(streams genericclioptions.IOStreams) *CRD2APIResourceSchemaOptions {
+	return &CRD2APIResourceSchemaOptions{
+		Options: base.NewOptions(streams),
+		Logs:    logs.NewOptions(),
+		Print:   genericclioptions.NewPrintFlags("crd2apiresourceschema"),
+	}
+}
+
+func (b *CRD2APIResourceSchemaOptions) AddCmdFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+	logsv1.AddFlags(b.Logs, cmd.Flags())
+	b.Print.AddFlags(cmd)
+}
+
+func (b *CRD2APIResourceSchemaOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run starts the process of converting CRDs to APIResourceSchema objects.
+func (b *CRD2APIResourceSchemaOptions) Run(ctx context.Context) error {
+	config, err := b.Options.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	client, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create dynamic client: %w", err)
+	}
+
+	apiResourceSchemaGVR := kubebindv1alpha2.SchemeGroupVersion.WithResource("apiresourceschemas")
+	crdGVR := apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions")
+	crdList, err := client.Resource(crdGVR).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list CRDs: %w", err)
+	}
+
+	for _, crd := range crdList.Items {
+		crdObj := &apiextensionsv1.CustomResourceDefinition{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(crd.UnstructuredContent(), crdObj); err != nil {
+			return fmt.Errorf("failed to convert CRD: %w", err)
+		}
+
+		if crdObj.Spec.Group == "kube-bind.io" {
+			fmt.Fprintf(b.Options.ErrOut, "Skipping CRD %s: belongs to group kube-bind.io\n", crdObj.Name)
+			continue
+		}
+
+		apiResourceSchema, err := convertCRDToAPIResourceSchema(crdObj, "apiresourceschema")
+		if err != nil {
+			fmt.Fprintf(b.Options.ErrOut, "Failed to convert CRD %s to APIResourceSchema: %v\n", crdObj.Name, err)
+		}
+
+		if apiResourceSchema == nil {
+			fmt.Fprintf(b.Options.ErrOut, "Skipping CRD %s: no schema found\n", crdObj.Name)
+			continue
+		}
+
+		unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(apiResourceSchema)
+		if err != nil {
+			return fmt.Errorf("failed to convert APIResourceSchema to unstructured: %w", err)
+		}
+		unstructuredResource := &unstructured.Unstructured{Object: unstructuredObj}
+
+		_, err = client.Resource(apiResourceSchemaGVR).Create(ctx, unstructuredResource, metav1.CreateOptions{})
```

Comment:
This creates in the cluster. We should just write it to files, like apigen in kcp does. 
so maybe `--output-dir` and write there. Applying to a cluster might not be desirable.

Reply:
Sorry, that's how I interpreted `create these objects from CRDs inside the cluster` 😄 
I will add both options as flags, although